### PR TITLE
plex: use buildFHSUserEnv; update to 1.13.9.5439

### PR DESCRIPTION
**Pull request by colemickens**
**NOTE**: Credit for this work and change all goes to @yegortimoshenko who actually did this module rewrite. I've just rebased it.

###### Motivation for this change

The original motivation is that Plex now downloads plugins at runtime and expects to be able to randomly execute such binaries.

* Replaces #38767  plex: use buildFHSUserEnv #38767 
* is rebased (albeit rebased and tested on `channels/nixos-unstable`)
* is tested with and without the module rewrite (note, I didn't test any sort of downgrade scenarios, only w/ and w/o the `buildFHSUserEnv` rewrite. according to discussion on #48505, this is sufficient)
* is updated with the latest (non-PlexPass) version of Plex

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by lukateras — APPROVED** [review]
This breaks reverse compatibility though, which is why I was hesitant to merge it.

**Comment by colemickens** [comment]
Is it the plugin stuff that is not backward/reverse compatible? I had tried upgrading to your rewritten module, and then removing your changes and Plex was still able to startup normally and had my library contents. Otherwise, I'm not sure I understand the problem.

**Comment by yegortimoshenko** [comment]
I think at its current form local SQLite DB is not reverse-compatible, i.e. you will have to relogin. I might be wrong, though.

**Comment by colemickens** [comment]
I'm not sure I understand why the data would be affected by the module changes. There's two things I can think about:
* the module rewrite - but it still stores data in the same place, so I don't see why it would be impacted?
* a plex upgrade that changes the DB in a way that isn't backward compatible -- given that we don't have any testing, this could have happened in the past, or maybe will never happen. It seems separate from the module rewrite.

The database also stores the media metadata. It seemed to work with and without your changes when I tested it - forward and backward. And if there were a login/logout, maybe it was the systemd unit restart (stores cookies in mem?)? If that's the only side effect, it seems small to me.

**Comment by yegortimoshenko** [comment]
> It seemed to work with and without your changes when I tested it - forward and backward.

Oh, OK, meaning I was wrong about SQLite DB. I thought it's now being stored in a different path.

Let's merge then.

**Comment by colemickens** [comment]
@yegortimoshenko So, I accidentally dropped your fix for the removed options, was trying to recreate it, and now I understand what you're concerned about.

I don't really understand how the Plex library plugins work, but it looks like the symlinking of the "skeleton directory" was customizable previously via the `dataDir` variable that is now removed. (I can't tell if it actually was customizing the location of the Plex runtime data though.)

Unfortunately, I am less confident that this won't break things if someone was using `dataDir` to change the plex data location, now that I realize this was customizable.

If it was just as simple as redirecting the entire `/var/lib/plex` directory, then I assume we can continue to support an option like that with the new `buildFHSUserEnv`. Let me know if you have suggestions and I can try to implement and test various upgrade/downgrades with my library in a custom `dataDir`.

(Also, if you have a branch with the renamed/removed options, can you repush it? Thanks.)

**Comment by dywedir** [comment]
Just merged update to 1.13.9.5456 https://github.com/NixOS/nixpkgs/pull/49715

**Comment by Infinisil** [comment]
The NixOS options need to be marked as removed, see `mkRemovedOptionModule`

**Comment by colemickens** [comment]
@andrew-d Some context from my *very* fuzzy recollection:
1. @yegortimoshenko had pushed some commits that I accidentally overwrote. It's been a while but I think it had to do with properly deprecating options I'd removed.
2. When I tried to reproduce it, I finally understood why there were concerns about this breaking people who were using the dataDir option previously, though I can't re-articulate them now.

Basically, I've been carrying this PR locally for months. I'd love to see it merged as well, but I need someone who uses dataDir option to weigh in, or I need someone with more weight to say that we can merge and deal with consequences later.

**Comment by bachp** [comment]
I think this can be closed not that https://github.com/NixOS/nixpkgs/pull/56565 is merged
